Source/JavaScriptCore/GLib.cmake

```diff
@@ -27,21 +30,30 @@ list(APPEND JavaScriptCore_INTERFACE_INCLUDE_DIRECTORIES
     "${JavaScriptCoreGLib_DERIVED_SOURCES_DIR}"
 )
 
+set(JavaScriptCore_HEADER_TEMPLATES
+    ${JAVASCRIPTCORE_DIR}/API/glib/JSCClass.h.in
+    ${JAVASCRIPTCORE_DIR}/API/glib/JSCContext.h.in
+    ${JAVASCRIPTCORE_DIR}/API/glib/JSCDefines.h.in
+    ${JAVASCRIPTCORE_DIR}/API/glib/JSCException.h.in
+    ${JAVASCRIPTCORE_DIR}/API/glib/JSCValue.h.in
+    ${JAVASCRIPTCORE_DIR}/API/glib/JSCVirtualMachine.h.in
+    ${JAVASCRIPTCORE_DIR}/API/glib/JSCWeakValue.h.in
+    ${JAVASCRIPTCORE_DIR}/API/glib/jsc.h.in
+)
+
 set(JavaScriptCore_INSTALLED_HEADERS
     ${JAVASCRIPTCORE_DIR}/API/glib/JSCAutocleanups.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/JSCClass.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/JSCContext.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/JSCDefines.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/JSCException.h
     ${JAVASCRIPTCORE_DIR}/API/glib/JSCOptions.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/JSCValue.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/JSCVirtualMachine.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/JSCWeakValue.h
-    ${JAVASCRIPTCORE_DIR}/API/glib/jsc.h
 
     ${JavaScriptCoreGLib_DERIVED_SOURCES_DIR}/jsc/JSCVersion.h
 )
 
+GENERATE_GLIB_API_HEADERS(JavaScriptCore JavaScriptCore_HEADER_TEMPLATES
+    ${JavaScriptCoreGLib_DERIVED_SOURCES_DIR}/jsc
+    JavaScriptCore_INSTALLED_HEADERS
+    "-DENABLE_2022_GLIB_API=$<BOOL:${ENABLE_2022_GLIB_API}>"
+)
+
```

Comment:
Could we split the patch and do the jsc in a different patch? it changes more things like autogenerating the headers.

Reply:
Sure, I can split the JSC API changes in a separate patch.